File: Hafta 1/credit.c
#include <stdio.h>
#include <cs50.h>

int kac_haneli(long card,int* character){
    int hane = 0;
    while (card > 0){
        if (card > 10 && card < 100) *character = card;
        hane++; card /= 10;
    }
    return hane;
}

bool kontrol(long card){
    int a =0, b=0;
    while (card > 0){
        a += card % 10;
        card /= 10;
        if (card > 0){
            int c = (card % 10)*2;
            b += c/10;
            b += c%10;
            card /= 10;
        }
    }
    int kalan = (a+b)%10;
    if (kalan == 0) return true;
    return false;
}

int main(){

    long card = get_long("kredi karti numaranizi giriniz : ");
    if (card <= 0) card = get_long("lutfen gecerli bir kredi karti numarasi giriniz : ");
    int character = 0;
    int hane = kac_haneli(card,&character);
    if (hane < 13 || hane > 16 || hane == 14 || !kontrol(card)) printf("INVALID\n");
    else{
        if (character == 34 && hane == 15) printf("AMERICAN EXPRESS\n");
         if (character == 37 && hane == 15) printf("AMEX\n");
        else if ((character / 10 == 4) && (hane == 13 || hane == 16)) printf("VISA\n");
        else if ((character == 51 || character == 52 ||character == 53 ||character ==54 || character ==55) && hane == 16) printf("MASTERCARD\n");
        else printf("INVALID\n");
    }

}

File: Hafta 6/credit.py
def card_control(card_number):
    a, b = 0, 0
    flag = True
    for i in range(len(card_number)-1, -1, -1):
        if (flag):
            a += int(card_number[i])
            flag = False
        else:
            c = int(card_number[i])*2
            b += c // 10
            b += c % 10
            flag = True
    return (a+b)%10 == 0


card_number = input("kart numaranizi giriniz : ")
if card_number[0] == "-":
    card_number = input("gecerli bir kart numarasi giriniz : ")
len_card = len(card_number)

if (len_card < 13 or len_card > 16 or len_card == 14 or not card_control(card_number)):
    print("INVALID")
else:
    character = int(card_number[0:2])
    if character >= 50:
        print("MASTERCARD")
    elif character >= 40:
        print("VISA")
    elif character == 37:
        print("AMEX")
    else:
        print("AMERICAN EXPRESS")
